### DISCUSSION TITLE
Is it safe to use *pg.DB from multiple GO routines?

### DISCUSSION THREAD

**Discussion by kulak** [go-pg/pg#2005] [Q&A] [answered]
The answer is probably Yes, but I'd like to make sure.

**Comment by elliotcourant** [comment, marked as answer]
Yes it is, but a transaction is not.

Queries run from the DB instance allocate a connection from the pool and execute the query and return the result synchronously and functions on the DB instance can be called from multiple go routines safely.

Transactions however are different in that it has a single dedicated connection/session, and can only be used by a single go routine.
